flink-core/src/main/java/org/apache/flink/configuration/ConfigConstants.java

```diff
@@ -243,6 +243,20 @@
 	 */
 	public static final String YARN_PROPERTIES_FILE_LOCATION = "yarn.properties-file.location";
 
+	/**
+	 * Prefix for passing custom environment variables to Flink's ApplicationMaster (JobManager).
+	 * For example for passing LD_LIBRARY_PATH as an env variable to the AppMaster, set:
+	 * 	yarn.application-master.env.LD_LIBRARY_PATH: "/usr/lib/native"
+	 * in the flink-conf.yaml.
+	 */
+	public static final String YARN_APPLICATION_MASTER_ENV_PREFIX = "yarn.application-master.env.";
+
+	/**
+	 * Similar to the {@see YARN_APPLICATION_MASTER_ENV_PREFIX}, this configuration prefix allows
+	 * setting custom environment variables.
+	 */
+	public static final String YARN_TASK_MANAGER_ENV_PREFIX = "yarn.taskmanager.env.";
```

Comment:
I wonder about the naming here. Maybe this should be `YARN_RESOURCE_MANAGER_ENV_PREFIX`? Or change `YARN_APPLICATION_MASTER_ENV_PREFIX` to `YARN_JOB_MANAGER_ENV_PREFIX`?